lib/ansible/modules/web_infrastructure/pmp.py

```diff
@@ -0,0 +1,296 @@
+        self.data['accountId'] = account['ACCOUNT ID']
+        return self.getAccountPwd()
+
+    def createResource(self, resourcename, accountname, password, ownername, resource_type='Linux', group=''):
+        data = {
+            "operation": {
+                "Details": {
+                    "RESOURCENAME": resourcename,
+                    "ACCOUNTNAME": accountname,
+                    "RESOURCETYPE": resource_type,
+                    "PASSWORD": password,
+                    "OWNERNAME": ownername,
+                }
+            }
+        }
+        if group:
+            data["operation"]["Details"]["RESOURCEGROUPNAME"] = group
+        headers = {
+            'Content-Type': 'text/json',
+        }  # admin test
+
+        url = self.format('https://{HOST}:{PORT}/restapi/json/v1/resources?AUTHTOKEN={TOKEN}')
+        data = json.dumps(data, sort_keys=False)
+        params = urllib.urlencode({'INPUT_DATA': data})
+        r = open_url(url, method="POST", headers=headers, data=params, validate_certs=self.validate_certs, use_proxy=self.use_proxy)
+        res = json.loads(r.read())['operation']['result']
+        status = res['status']
+        if status == 'Failed':
+            raise RuntimeError(res['message'])
+        return status
+
+
+def get_password(auth_pmp, params):
+    pmp = PasswordManagerPro(auth_pmp)
+    password = pmp.getPassword(params['resource_name'], params['account_name'])
+    return {'status': 'OK', 'password': password}
+
+
+def create_host(auth_pmp, params):
+    pmp = PasswordManagerPro(auth_pmp)
+    status = pmp.createResource(params['resource_name'], params['account_name'], params['password'], params['owner'],
+                                group=params['group'])
+    return {'status': status}
+
+
+# Some parameters are required depending on the operation:
+OP_REQUIRED = dict(create_host=['resource_name', 'account_name', 'password', 'owner'],
+                   get_password=['resource_name', 'account_name'],
+                   )
+
+
+def main():
+    global module
+    module = AnsibleModule(
+        argument_spec=dict(
+            uri=dict(required=True),
+            port=dict(required=False, default="7272"),
+            operation=dict(choices=['get_password', 'create_host'],
+                           aliases=['command'], required=True),
+            token=dict(required=True, no_log=True),
+            resource_name=dict(),
+            account_name=dict(),
+            password=dict(no_log=True),
+            owner=dict(),
+            group=dict(),
+            use_proxy=dict(default=False),
+            validate_certs=dict(default=False),
+            timeout=dict(type='float', default=10),
+        ),
+        supports_check_mode=False
+    )
+
+    op = module.params['operation']
+    # Check we have the necessary per-operation parameters
+    missing = []
+    for parm in OP_REQUIRED[op]:
+        if not module.params[parm]:
+            missing.append(parm)
```

Comment:
You can use ```required_if``` which is built into AnsibleModule instead of having your own code.  Look at https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/cloudstack/cs_storage_pool.py#L489  as a random example of using that.